core/src/domain/listeners/leaseInspectionCreationOnLeaseCreation.listener.ts

```diff
@@ -0,0 +1,21 @@
+import { LeaseInspectionService } from '@domain/services/leaseInspection.service';
+import { LeaseInspectionCreationOnLeaseCreationEvent } from '@domain/events/leaseInspectionCreationOnLeaseCreation.event';
+import { Injectable } from '@nestjs/common';
+import { OnEvent } from '@nestjs/event-emitter';
+import { LeaseInspectionStateDto, LeaseInspectionTypeDto } from '@infrastructure/dtos/enum/leaseInspection.enum.dto';
+import { LeaseInspectionDto } from '@infrastructure/dtos/leaseInspection.dto';
+
+@Injectable()
+export class LeaseInspectionCreationOnLeaseCreationListener {
+    constructor(
+        private readonly leaseInspectionService: LeaseInspectionService,
+    ) {
+    }
+
+    @OnEvent(LeaseInspectionCreationOnLeaseCreationEvent.eventName, { async: true })
+    async handleElementCreatedOnLeaseInspectionCreationEvent(event:LeaseInspectionCreationOnLeaseCreationEvent ) {
```

Comment:
here you are waiting for a LeaseInspectionCreationOnLeaseCreationEvent  as param